crates/client/db/src/mapping_db.rs

```diff
@@ -85,14 +88,22 @@ impl<B: BlockT> MappingDb<B> {
 
         transaction.set(crate::columns::SYNCED_MAPPING, &commitment.block_hash.encode(), &true.encode());
 
-        for transaction_hash in commitment.starknet_transaction_hashes.into_iter() {
+        for transaction_hash in commitment.starknet_transaction_hashes.iter() {
```

Comment:
moves
```rust
if commitment.cache {
            transaction.set(
                crate::columns::STARKNET_TRANSACTION_HASHES_CACHE,
                &commitment.starknet_block_hash.encode(),
                &commitment.starknet_transaction_hashes.encode(),
            );
        }
```

before your loop and you will retain the ability to iter by value rather than ref

Reply:
Why do you want to iter by value? The value gets encoded anyway and `encode` takes a shared reference. I'm pretty sure that iterating by value is slower anyway because of the `H256` copies. (to be fair, both are probably optimized the same but heh)

Reply:
no specific reason. And your are right that it won't change the perf